Makefile

```diff
@@ -19,37 +36,92 @@ ALL_IMAGES:=base-notebook \
 
+
+
 build/%: DARGS?=
-build/%: ## build the latest image for a stack
-	docker build $(DARGS) --rm --force-rm -t $(OWNER)/$(notdir $@):latest --build-arg OWNER=$(OWNER) ./$(notdir $@)
+build/%: ## build the latest image for a stack on amd64 only
+	@echo "::group::Build $(OWNER)/$(notdir $@) (amd64)"
+	docker build $(DARGS) --rm --force-rm -t $(OWNER)/$(notdir $@):latest ./$(notdir $@) --build-arg OWNER=$(OWNER)
 	@echo -n "Built image size: "
 	@docker images $(OWNER)/$(notdir $@):latest --format "{{.Size}}"
-
+	@echo "::endgroup::Build $(OWNER)/$(notdir $@) (amd64)"
 build-all: $(foreach I,$(ALL_IMAGES), build/$(I) ) ## build all stacks
-build-test-all: $(foreach I,$(ALL_IMAGES), build/$(I) test/$(I) ) ## build and test all stacks
+
+# Limitations on docker buildx build (using docker/buildx 0.5.1):
+#
+# 1. Can't --load and --push at the same time
+#
+# 2. Can't --load multiple platforms
+#
+# What does it mean to --load?
+#
+# - It means that the built image can be referenced by `docker` CLI, for example
+#   when using the `docker tag` or `docker push` commands.
+#
+# Workarounds due to limitations:
+#
+# 1. We always make a dedicated amd64 build named as OWNER/<stack>-notebook so
+#    we always can reference that image no matter what.
+#
+# 2. We always also build a multi-platform image during build-multi that will be
+#    inaccessible with `docker tag` and `docker push` etc, but this will help us
+#    test the build on the different platform and provide cached layers for
+#    later.
+#
+# 3. We let push-multi refer to rebuilding a multi image with `--push`.
+#
+#    We now rely on the cached layer.
+#
+# Outcomes of the workaround:
+#
+# 1. We can keep using the previously defined Makefile commands that doesn't
+#    include `-multi` suffix as before.
+#
+# 2. Assuming we have setup docker/dockerx properly to build in arm64
+#    architectures as well, then we can build and publish such images via the
+#    `-multi` suffix without needing a local registry.
+#
+build-multi/%: DARGS?=
+build-multi/%: ## build the latest image for a stack on amd64 and arm64
+	@echo "::group::Build $(OWNER)/$(notdir $@) (amd64)"
+	docker buildx build $(DARGS) --rm --force-rm -t $(OWNER)$(notdir $@):latest ./$(notdir $@) --build-arg OWNER=$(OWNER) --platform "linux/amd64" --load
+	@echo -n "Built image size: "
+	@docker images $(OWNER)/$(notdir $@):latest --format "{{.Size}}"
+	@echo "::endgroup::Build $(OWNER)/$(notdir $@) (amd64)"
+
+	@echo "::group::Build $(OWNER)/$(notdir $@) (amd64,arm64)"
+	docker buildx build $(DARGS) --rm --force-rm -t build-multi-tmp-cache/$(notdir $@):latest ./$(notdir $@) --build-arg OWNER=$(OWNER) --platform "linux/amd64,linux/arm64"
+	@echo "::endgroup::Build $(OWNER)/$(notdir $@) (amd64,arm64)"
+build-all-multi: $(foreach I,$(MULTI_IMAGES), build-multi/$(I)) $(foreach I,$(AMD64_ONLY_IMAGES), build/$(I)) ## build all stacks
```

Comment:
This should be called build-all, because you're also building non-multi images.

Reply:
I want to have two separate options, one `build-all` like we have had it, doing exactly what it did - and `build-all-multi` for example like above, where it builds all in as many archs it can.